services/s3/src/main/java/software/amazon/awssdk/services/s3/multipart/MultipartConfiguration.java

```diff
@@ -0,0 +1,206 @@
+     * @return the value of the configured maximum memory usage.
+     */
+    public Long apiCallBufferSizeInBytes() {
+        return this.apiCallBufferSizeInBytes;
+    }
+
+    /**
+     * Builder for a {@link MultipartConfiguration}.
+     */
+    public interface Builder extends CopyableBuilder<Builder, MultipartConfiguration> {
+
+        /**
+         * Configures the minimum number of bytes of the body of the request required for requests to be converted to their
+         * multipart equivalent. Only taken into account when converting {@code putObject} and {@code copyObject} requests.
+         * Any request whose size is less than the  configured value will not use multipart operation,
+         * even if multipart is enabled via {@link S3AsyncClientBuilder#multipartEnabled(Boolean)}.
+         * <p></p>
+         *
+         * Default value: 8 Mib
+         *
+         * @param thresholdInBytes the value of the threshold to set.
+         * @return an instance of this builder.
+         */
+        Builder thresholdInBytes(Long thresholdInBytes);
+
+        /**
+         * Indicates the value of the configured threshold.
+         * @return the value of the threshold.
+         */
+        Long thresholdInBytes();
+
+        /**
+         * Configures the part size, in bytes, to be used in each individual part requests.
+         * <p></p>
+         * When uploading large payload, the size of the payload of each individual part requests might actually be
+         * bigger than
+         * the configured value since there is a limit to the maximum number of parts possible per multipart request. If the
+         * configured part size would lead to a number of parts higher than the maximum allowed, a larger part size will be
+         * calculated instead to allow fewer part to be uploaded, to avoid the limit imposed on the maximum number of parts.
+         * <p></p>
+         * In the case where the {@code minimumPartSizeInBytes} is set to a value higher than the {@code thresholdInBytes}, when
+         * the client receive a request with a size smaller than a single part multipart operation will <em>NOT</em> be performed
+         * even if the size of the request is larger than the threshold.
+         * <p></p>
+         * Default value: 8 Mib
+         *
+         * @param minimumPartSizeInBytes the value of the part size to set
+         * @return an instance of this builder.
+         */
+        Builder minimumPartSizeInBytes(Long minimumPartSizeInBytes);
+
+        /**
+         * Indicated the value of the part configured size.
+         * @return the value of the part size
+         */
+        Long minimumPartSizeInBytes();
+
+        /**
+         * Configures the maximum amount of memory, in bytes, the SDK will use to buffer content of requests in memory.
+         * Increasing this value may lead to better performance at the cost of using more memory.
+         * <p></p>
```

Comment:
Nit: I generally see people just do the (invalid XHTML) `<p>` between paragraphs or wrap the non-leading paragraphs in `<p>` and `</p>` I don't usually see this pattern. It probably works, but it looks weird to me.

Reply:
It's in our guideline :) https://github.com/aws/aws-sdk-java-v2/blob/master/docs/design/APIReference.md

`A single <p> MUST be used between paragraphs.`